src/parquet/file/reader-internal.cc

```diff
@@ -268,15 +259,10 @@ void SerializedFile::ParseMetaData() {
     throw ParquetException("Invalid parquet file. File is less than "
         "file metadata size.");
   }
-
   source_->Seek(metadata_start);
 
   std::vector<uint8_t> metadata_buffer(metadata_len);
-  bytes_read = source_->Read(metadata_len, &metadata_buffer[0]);
-  if (bytes_read != metadata_len) {
-    throw ParquetException("Invalid parquet file. Could not read metadata bytes.");
-  }
-
+  source_->Read(metadata_len, &metadata_buffer[0]);
```

Comment:
We have to keep the check here then ? since `bytes_read` could hypothetically be less than `metadata_len`